src/httpcommon.cpp

```diff
@@ -192,6 +192,28 @@ namespace http {
     return 0;
   }
 
+  int
+  check_url_response(std::string url) {
+    int response_code = 0;
+    CURL *curl = curl_easy_init();
+    if (curl) {
+      CURLcode result;
+      curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+      curl_easy_setopt(curl, CURLOPT_HTTPGET, 1);  // nginxproxymanager responds to HEAD requests with 502
+      curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0);  // ignore self-signed cert
+      curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);  // ignore self-signed cert
```

Comment:
These lines will let the function detect a proxied or other address that is still using a self-signed cert. For example, you can test functionality without a reverse proxy by setting the url to `https://sunshinehost:47990`.

It might be OK to leave this here, but if we want to enforce valid SSL checking, I can remove the lines and perhaps add a note that a valid SSL cert will be required to the docs.